rewrite-pattern.js

```diff
@@ -106,7 +108,10 @@ const getUnicodePropertyEscapeSet = (value, isNegative) => {
 	}
 	return {
 		characters: set.characters.clone(),
-		strings: new Set(set.strings || [])
+		strings: set.strings
+			// We need to escape strings like *️⃣ to make sure that they can be safelu used in unions
+			? new Set(set.strings.map(str => SPECIAL_CHARS.has(str[0]) ? `\\${str}` : str))
```

Comment:
I quickly approved this from my phone last night to resolve the immediate issue, but I think a more robust fix would be to escape special characters at any position in each string. There could in theory, or in the future, be a properties of strings that expands to `a`, `a|`, and `a*`, for example.

Reply:
Filed it here: https://github.com/mathiasbynens/regexpu-core/issues/64